node/actors/control_api.go

```diff
@@ -199,44 +291,107 @@ func (api *ControlAPI) handleInfo(m *nats.Msg) {
-		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("failed to get list of running workloads: %s", err))
+		api.logger.Error("Failed to get node info", slog.Any("error", err))
+		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("failed to get node info: %s", err))
 		return
 	}
-	workloadResponse, ok := response.(*actorproto.WorkloadListing)
+
+	workloadResponse, ok := response.(*actorproto.NodeInfo)
 	if !ok {
 		api.logger.Error("Workload listing response from agent supervisor was not the correct type")
 		respondEnvelope(m, InfoResponseType, 500, nil, "Agent supervisor returned the wrong data type")
 		return
 	}
 
-	// TODO: This struct will be replaced by a generated struct from a JSON schema in api/nodecontrol
-	info := struct {
-		NodeID    string                        `json:"node_id"`
-		Workloads []*actorproto.WorkloadSummary `json:"workloads"` // NOTE: the json schema gernerated version of this won't use actorproto
-	}{
-		NodeID:    "Nxxx",
-		Workloads: workloadResponse.Workloads,
-	}
-	respondEnvelope(m, InfoResponseType, 200, info, "")
+	respondEnvelope(m, InfoResponseType, 200, infoResponseFromProto(workloadResponse), "")
 }
 
 func (api *ControlAPI) handleLameDuck(m *nats.Msg) {
-	// TODO
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, LameDuckResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+	// Ask the agent supervisor for a list of all the workloads from all of its children
+	response, err := api.self.Ask(ctx, agentSuper, new(actorproto.SetLameDuck))
+	if err != nil {
+		api.logger.Error("Failed to put node in lame duck mode", slog.Any("error", err))
+		respondEnvelope(m, LameDuckResponseType, 500, nil, fmt.Sprintf("failed to put node in lame duck mode: %s", err))
+		return
+	}
+
+	workloadResponse, ok := response.(*actorproto.LameDuckResponse)
+	if !ok {
+		api.logger.Error("Workload listing response from agent supervisor was not the correct type")
+		respondEnvelope(m, LameDuckResponseType, 500, nil, "Agent supervisor returned the wrong data type")
+		return
+	}
+
+	respondEnvelope(m, LameDuckResponseType, 200, lameDuckResponseFromProto(workloadResponse), "")
 }
 
 func (api *ControlAPI) handlePing(m *nats.Msg) {
-	// TODO
-}
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, LameDuckResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
 
-func (api *ControlAPI) handleUndeploy(m *nats.Msg) {
-	// TODO
+	response, err := api.self.Ask(ctx, agentSuper, new(actorproto.PingNode))
```

Comment:
pingnode shouldn't be something we ask the agent supervisor for. The control API should return a ping response, which may contain information obtained from the agent supervisor. Ideally it's probably coming from the same "Get workloads" query that the agent supervisor handles for node info.